src/main/java/com/esri/samples/map/mobile_map_search_and_route/MobileMapSearchAndRouteSample.java

```diff
@@ -0,0 +1,316 @@
+import com.esri.arcgisruntime.tasks.networkanalysis.RouteTask;
+import com.esri.arcgisruntime.tasks.networkanalysis.Stop;
+
+public class MobileMapSearchAndRouteSample extends Application {
+
+  private MapView mapView;
+  private MobileMapPackage mobileMapPackage;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Mobile Map Search and Route Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a map view
+      mapView = new MapView();
+
+      // create a list view to list the maps in a mobile map package
+      ListView<ArcGISMap> mapPackageListView = new ListView<>();
+      mapPackageListView.setMaxSize(170, 100);
+
+      // create a file chooser to find local mmpk files
+      FileChooser fileChooser = new FileChooser();
+      fileChooser.setInitialDirectory(new File("./samples-data/mmpk"));
+      fileChooser.setInitialFileName(fileChooser.getInitialDirectory().getAbsolutePath() + "/SanFrancisco.mpk");
+      FileChooser.ExtensionFilter mpkFilter = new FileChooser.ExtensionFilter("Map Packages (*.mmpk)", "*.mmpk");
+      fileChooser.getExtensionFilters().add(mpkFilter);
+
+      // click a button to open the file chooser
+      Button findMmpkButton = new Button("Open mobile map package");
+      findMmpkButton.setOnAction(e -> {
+        File selectedMmpk = fileChooser.showOpenDialog(mapView.getScene().getWindow());
+        if (selectedMmpk != null) {
+          // remember the directory of the mmpk for next time
+          fileChooser.setInitialDirectory(selectedMmpk.getParentFile());
+          fileChooser.setInitialFileName(selectedMmpk.getAbsolutePath());
+
+          // clear the map and list selection from any previous mmpks
+          mapPackageListView.getSelectionModel().clearSelection();
+          mapPackageListView.getItems().clear();
+          mapView.setMap(null);
+
+          // create a mobile map package from the file path and load it
+          mobileMapPackage = new MobileMapPackage(selectedMmpk.getAbsolutePath());
+          mobileMapPackage.loadAsync();
+          mobileMapPackage.addDoneLoadingListener(() -> {
+            if (mobileMapPackage.getLoadStatus() == LoadStatus.LOADED && !mobileMapPackage.getMaps().isEmpty()) {
+              // show the maps belonging to the mobile map package in the list view
+              mobileMapPackage.getMaps().forEach(map -> mapPackageListView.getItems().add(map));
+              // default to displaying the first map in the map view
+              mapPackageListView.getSelectionModel().select(mobileMapPackage.getMaps().get(0));
+            } else {
+              new Alert(Alert.AlertType.ERROR, "Failed to load the mobile map package").show();
```

Comment:
Would add in the alert message that it may not be working because MMPK didn't have and maps within it. Could also break this down into two different alerts.